agents/postgres/parser/parser.go

```diff
@@ -0,0 +1,109 @@
+package parser
+
+import (
+	"reflect"
+
+	pgquery "github.com/lfittl/pg_query_go"
+	pgquerynodes "github.com/lfittl/pg_query_go/nodes"
+)
+
+// Parser parses postgresql queries
+type Parser struct{}
+
+// ExtractTables extracts table names from query.
+func (p *Parser) ExtractTables(query string) (tables []string, err error) {
```

Comment:
I'm not sure, but maybe it will be better don't use named return parameters here?